packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -475,11 +478,11 @@ export class NftDetectionController extends StaticIntervalPollingControllerV1<
       const { selectedAddress: previouslySelectedAddress, disabled } =
         this.config;
 
-      if (
-        selectedAddress !== previouslySelectedAddress ||
-        !useNftDetection !== disabled
-      ) {
-        this.configure({ selectedAddress, disabled: !useNftDetection });
+      if (selectedAddress !== previouslySelectedAddress) {
+        this.configure({ selectedAddress });
+      }
+      if (!useNftDetection !== disabled) {
+        this.configure({ disabled: !useNftDetection });
```

Comment:
I'm wondering if this logic should stay like before.  In the case of account switch, I think we probably do want a detection to trigger.  But it doesn't look like the new code would do this, and I think the old code did

Reply:
Having the logic like before would trigger the `start()` of the detection when the user closes browser all together then opens it without opening MM. This is happening because of the `OR` statement in the logic where initially the selectedAddress is set to '' in the `defaultConfig` , which makes this statement `selectedAddress !== previouslySelectedAddress` evaluate to `true` so it goes in and starts fetching from OS. (Then the selected address will be the correct address rather than '' but it would be too late cause the start() was already triggered.

It is true that doing the logic in this PR will not trigger the start() immediately when the user switches the account, but because the nftDetection is already ON, the next time its doing the detection (3mins) it will detect against the correct address.

I think that this might be okay with the 3mins interval, but in the future when we make this interval larger, this wont be an optimal user experience.

We can pass the selectedAddress in the constructor like the disabled? rather than doing 
```
    this.defaultConfig = {
      interval: DEFAULT_INTERVAL,
      chainId: initialChainId,
      selectedAddress: '',
      disabled: initialDisabled,
    };
```
   it will be 
```
       this.defaultConfig = {
      interval: DEFAULT_INTERVAL,
      chainId: initialChainId,
      selectedAddress: initialSelectedAddress,
      disabled: initialDisabled,
    };
```
That should avoid having "" for a brief moment

Reply:
I think when we make those changes ☝️  we can have the old OR statement logic, and it should not trigger polling when browser is re-opened without opening MM.  Can work on this fix, but lemme know if anything 🙏

Reply:
I think that fix makes sense if it works.  If not I could also live with the 3 minute gap, since the unncessesary background polling is more important to fix

Reply:
🙏  fixed eventually for future updates of this interval 🙏